### ISSUE TITLE
Consider adding a ‘Bed Broken’ Advancement

### ISSUE THREAD

**Issue by AllYourBlocks** [AllYourBlocks/ChunkWars#25]
This deviates from the as-vanilla-as-possible concept, but could *feel* like vanilla as it would read out in game chat.

It could create some interesting situations in game where e.g. Yellow Bed is broken stealthily by Blue, but with Yellow Players on the Chunk in close proximity to respond.

Would need to be per-bed, so it triggers only when a Yellow Player breaks Blue's Bed, or when a Blue Player breaks Yellow's Bed.

**Comment by AllYourBlocks** [comment]
Could also play a sound

**Comment by AllYourBlocks** [comment]
Idea accepted and should be added as a toggle-able option in game.

Have added to the Options Panel already.

**Comment by slipcor** [comment]
We need to look into how to properly do this. I hope there is a "prerequisite" (I forgot the datapack-advancement-requirement name) for "custom score set", so that we can actually check whether the option is activated.

If it is, it will be pretty easy to implement :) breaking beds. duh. Well technically it shouuld ignore players breaking their own bed? Or... not? :D It has to be color specific maybe :( what are your thoughts? @AllYourBlocks

**Comment by AllYourBlocks** [comment]
It should ignore teams' breaking their own beds.

We need to think about and discuss whether it triggers only on the first bed break, or on every bed break (and which of these is possible to show as an achievement/advancement(?!) assuming that is the chosen display method)  and whether there are any technical limitations.

**Comment by slipcor** [comment]
If I can restrict the color by checking the player's team score, we can also restrict it by other arbitrary scores we can change at any time.
I think a good solution would be resetting the "other team player's advancement" in every tick that the bed of one team exists, because in a 4 team situation, there has to be a way to give this advancement for all three other colors basically at the same time if one team has all their people out to kill beds. Should be rare as all hell but can happen :)

It would not be the end of the world because one bed break announcement serves as an alert to basically everyone to assume they are broken?

Also another thing to consider, for that same indentifying reason is, should the advancement be given to the breaker or one of the remaining team members of the team which bed has been broken? There would be the visual indicator there then which bed has been broken.

Also we could manually give the team whose bed has been broken a title/subtitle indicator message :D

**Comment by AllYourBlocks** [comment]
Adding this would also take away from the surprise element of not having your own bed, and discovering this during the game. Great for observers, less good for players?!

**Comment by AllYourBlocks** [comment]
Agreed that we will add this as a toggle-able option and have discussed the intended behaviour for the renamed 'Bed Status' option.

Possible chat output to read;

[Player] has broken [Colour] Team's Bed
[Player] has replaced [Colour] Team's Bed

or

[Colour] Team's Bed has been broken by [Player]
[Colour] Team's Bed has been replaced by [Player]

**Comment by AllYourBlocks** [comment]
References #71

**Comment by AllYourBlocks** [comment]
Tested and works in 2-team mode, needs to be tested in 4-team.
